pkg/workflows/builder_test.go

```diff
@@ -0,0 +1,134 @@
+package workflows_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+)
+
+func NewWorkflowSpec() (workflows.WorkflowSpec, error) {
+	workflow := workflows.NewWorkflow(workflows.NewWorkflowParams{
+		Owner: "00000000000000000000000000000000000000aa",
+		Name:  "ccipethsep",
+	})
+
+	mercuryTriggerOutput := workflows.AddTrigger(workflow, triggers.NewMercuryTrigger(
+		triggers.NewMercuryTriggerParams{
+			Config: triggers.Config{
+				FeedIDs: []string{
+					"0x0003fbba4fce42f65d6032b18aee53efdf526cc734ad296cb57565979d883bdd",
+					"0x0003c317fec7fad514c67aacc6366bf2f007ce37100e3cddcacd0ccaa1f3746d",
+					"0x0003da6ab44ea9296674d80fe2b041738189103d6b4ea9a4d34e2f891fa93d12",
+				},
+				MaxFrequencyMs: 100,
+			},
+		},
+	))
+
+	fmt.Print("mercuryTriggerOutput", mercuryTriggerOutput)
+
+	// TODO: Input shows up as an interpolated value in the spec
+	// QQ: How is interpolated value represented today?
+	// A: workflows.StepInputs
+	// QQ: How does workflows.StepInputs look like?
+	// A: It has OutputRef and Mapping
+	// QQ: How would one represent inputs from multiple nodes to the consensus
+	// capability?
+	// A: Consensus capability can multiple the inputs for testing purposes.
+
+	consensusOutput := workflows.AddConsensus(workflow, ocr3.NewOCR3Consensus())
+
+	fmt.Print("consensusOutput", consensusOutput)
+
+	return workflow.Spec(), nil
+}
+
+func TestBuilder_ValidSpec(t *testing.T) {
+	testWorkflowSpec, err := NewWorkflowSpec()
+	require.NoError(t, err)
+
+	expectedSpec := workflows.WorkflowSpec{
```

Comment:
Expected output - this can be converted to YAML or used by workflow engine. Incomplete.